aws-lc-rs/src/aead/poly1305.rs

```diff
@@ -65,6 +65,7 @@ impl Context {
         unsafe {
             let mut tag = MaybeUninit::<[u8; TAG_LEN]>::uninit();
             CRYPTO_poly1305_finish(self.state.0.as_mut_ptr().cast(), tag.as_mut_ptr().cast());
+            crate::fips::set_fips_service_status_unapproved();
```

Comment:
The pattern seems to be to only set the "unapproved" indicator after the operation has successfully completed.  Would it be more "friendly" to (also) set it at at the start (in `from_key`) instead of waiting until they've completed the operation?

Reply:
I was opting to be similar in consistency to AWS-LC. So for example the indicator is set at the end of functions only when they are successful. Similarly for the indicators are I manually plumbed through to set to false, I opted to do it in similar spots where the indicator was set in AWS-LC for similar functionality.